torch/fx/passes/utils/matcher_utils.py

```diff
@@ -0,0 +1,158 @@
+from collections import defaultdict
```

Comment:
I'm expecting, somewhere here, a high level plain English explanation of the algorithm

Reply:
done.  
see match() function description.